trail/upload/process_uploads/fits_processor.py

```diff
@@ -0,0 +1,213 @@
+
+    An upload is determined to be a FITS file if its extensions ends on one
+    of the allowed extensions found in `extensions`.
+
+    Parameters
+    ----------
+    upload : `upload_wrapper.TemporaryUploadedFileWrapper`
+            Uploaded file.
+    """
+
+    extensions = [".fit", ".fits", ".fits.fz"]
+    """File extensions this processor can handle."""
+
+    def __init__(self, uploadInfo, uploadedFile):
+        super().__init__(uploadInfo, uploadedFile)
+        self.hdulist = fits.open(uploadedFile.tmpfile.temporary_file_path())
+        self.primary = self.hdulist["PRIMARY"].header
+        self.standardizer = HeaderStandardizer.fromHeader(self.primary,
+                                                          filename=uploadedFile.filename)
+        self.isMultiExt = len(self.hdulist) > 1
+
+    @staticmethod
+    def _isMultiExtFits(hdulist):
+        """Returns `True` when given HDUList contains more than 1 HDU.
+
+        Parameters
+        ----------
+        hdulist : `astropy.io.fits.HDUList`
+            An HDUList object.
+        """
+        return len(hdulist) > 1
+
+    @classmethod
+    def normalizeImage(cls, image):
+        """Normalizes the image data to the [0,1] domain, using histogram
+        equalization.
+
+        Returns
+        -------
+        norm : `np.array`
+            Normalized image.
+        """
+        # TODO: make things like these configurable (also see resize in
+        # store_thumbnail)
+        stretch = aviz.HistEqStretch(image)
+        norm = aviz.ImageNormalize(image, stretch=stretch, clip=True)
+
+        return norm(image)
+
+    @classmethod
+    def _createThumbnails(cls, filename, image, basewidth=640):
+        """Creates a large and a small thumbnail images normalized to [0, 255]
+        domain and their save locations.
+
+        Parameters
+        ----------
+        filename : `str`
+            Image filename.
+        image : `numpy.array`
+            Image
+        basewidth : `int`, optional
+            Width of the smaller thumbnail.
+
+        Returns
+        -------
+        largeThumb : `dict`
+            Dictionary containing the save location of the thumbnail,
+            `savepath`, and the image, `thumb`.
+        smallThumb : `dict`
+            Dictionary containing the save location of the thumbnail,
+            `savepath`, and the image, `thumb`.
+        """
+        normedImage = cls.normalizeImage(image)
+
+        # TODO: a note to fix os.path dependency when transitioning to S3
+        # and fix saving method from plt to boto3
+        smallPath = os.path.join(cls.media_root, filename+'_small.jpg')
+        largePath = os.path.join(cls.media_root, filename+'_large.jpg')
+
+        # TODO: consider removing PIL dependency once trail detection is
+        # implemented, if it is implemented via OpenCV
+        normedImage = (normedImage.data*255).astype(np.uint8)
+        # this is grayscale
+        img = Image.fromarray(normedImage, "L")
```

Comment:
AFAIK `matplotlib` can't be used to resize an image, so we need something else. Matplotlib will use PIL in the background anyhow, every time we [save an image ](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.imsave.html). 

PIL here is used just for resizing of the image, not its normalization, because it already comes with `matplotlib`. I guess we can transition to something like scikit image module for this but I wasn't sure if we want to install the entire scikit and scikit image extension for a single operation.
Eventually, if we implement some kind of image science data analysis stuff, I assume we will use a more powerful libraries like OpenCV or something. We can change this to cv resizing method then too. 

Resizing is not particularly slow, although there are magic keywords that can make it even faster. 

As far as image normalization goes: For all cases except DECam focal plane images, and even then only in cases when the file was created by the DECam Community Pipelines, the image normalization is performed by `Astropy.ImageNormalization` (see `FitsProcessor.normalizeImage`). 
We can change the normalization to use the asynh stretch, but that will only work well, for a particular choice of normalization constant, if we're processing calibrated exposures and will often not  be generally applicable, with the same normalization constant, for raw, just isr or fully calibrated image data. 

I picked histogram equalization because it's mathematically guaranteed to maximize contrast (which is what I thought I wanted when we are resizing down this much) but I'm not married to the idea. Again, see the comments in `FiitsProcessor.normalizeImage` method as it was left there as a potential `TODO` for anyone wanting to roll up their sleeves. 

As far as the DECam Community Pipeline image products - I'm not sure we can live with astropy's normalization there, see comments in `DecamFits.normalizeImage` processor method:

>      # astropy equalizer averages 4.3 seconds, the PIL approach can bring
>      # that down to cca 0.13. Without normalization the time is 0.04s

That's with the speedier histogram equalization, performance may take a larger hit.

Reply:
OK great points, PIL it is! I just hadn't seen it before; I guess in Rubin we have all our own image manipulation C++ flavored shenanigans.

Reply:
That's an excellent point about how the normalization I suggested may only work well for halfway calibrated images. We can leave it as histogram equalization to maximize contrast 👍

Reply:
Yeah, there could be a particular normalization factor for asinh that works well across the board and we could add the normalization factor as an attribute, or a property for those processors that have multiple different appropriate normalization factors, and it would work just fine. It will make the images look nicer. 

Is this resolved, pending an issue creation or should we make this a requirement for this PR?